### ISSUE TITLE
Infinity values at bounding-box when calculating the geo-bounds aggregate with many buckets

### ISSUE THREAD

**Issue by plroit** [elastic/elasticsearch#10804]
Hi,

I am following the example at: 
http://www.elastic.co/guide/en/elasticsearch/guide/current/geo-bounds-agg.html
to create an aggregation of some sample data according to their geohash cells, 
and return the geo-bounds of the result set in the bucket.
```
    "aggs": {
        "bucket" : {
            "geohash_grid" : {
                "field": "centroid",
                "precision": 8
            },
            "aggs": {
                "cell": {
                    "geo_bounds": {
                        "field": "centroid"
                    }
                }
            }
        }
    }
```
I have noticed that when my precision becomes higher the returned geo-bounds contain
infinity values for longitude and latitude. The bbox in every bounds object is of the following form: 
```
                  "bounds": {
                     "top_left": {
                        "lat": 49.611142831107415,
                        "lon": "-Infinity"
                     },
                     "bottom_right": {
                        "lat": "-Infinity",
                        "lon": "-Infinity"
                     }
                  }
```
(only top-left.lat changes)

Reducing the number of buckets by geo-filtering seems to help but is very limiting.
For example, I applied a geo-bounding-box filter with the bounds of a 
single geo-hash cell of precision 6, then geo-aggregated using precision 7,
with 32 buckets that returned fine. However geo-aggregating with precision 8
(returning 1024 buckets) have produced the incorrect geo-bounds.

The mapping that I've used:
```
            "properties": {
               "centroid": {
                  "type": "geo_point",
                  "lat_lon": true,
                  "geohash": true,
                  "geohash_prefix": true,
                  "geohash_precision": 11
               },
               "geom": {
                  "type": "geo_shape",
                  "tree_levels": 10
               },
               "name": {
                  "type": "string"
               },
               "osmId": {
                  "type": "long"
               },
               "type": {
                  "type": "string",
                  "index": "not_analyzed"
               }
            }
```

Thanks

**Comment by clintongormley** [comment]
Hi @plroit 

Would be very useful if you could provide some example docs, in fact a full simple recreation would be awesome.

thanks

/cc @colings86

**Comment by plroit** [comment]
Sorry for the delay,

The query that I have used:

```
POST /osm/building/_search
{
    "size": 0, 
    "aggs": {
        "bucket" : {
            "geohash_grid" : {
                "field": "centroid",
                "precision": 8
            },
            "aggs": {
                "cell": {
                    "geo_bounds": {
                        "field": "centroid"}}}}}}
```

The dataset comes from OpenStreetMap vector data of Luxemburg,
[You can find it here] (http://download.geofabrik.de/europe/luxembourg.html)
I am trying to put into geohash buckets  all the buildings, which are polygon shapes.
The field 'centroid' is the point center of the polygon.

Reviewing the indexed data reveals nothing special or erroneous.

**Comment by plroit** [comment]
I can provide you with a generated csv file of the 77K buildings, and the C# code that I have used for indexing them. Would that help?

**Comment by colings86** [comment]
@plroit the csv file would be very useful if you could provide that. Don't worry about the C# code for now.

**Comment by plroit** [comment]
Try this file, each line is the object serialized to json
https://www.dropbox.com/s/49i73lfe2ebdvml/building.csv?dl=0

**Comment by colings86** [comment]
@plroit thanks for the file and for raising this bug. I have reproduced the problem with the docs you provided so I'll look into what's causing this.

**Comment by colings86** [comment]
@plroit Thanks again for raising this, I have just merged the fix via #10917 and it should be available from version 1.5.3 onwards

**Comment by galkinsergeyv** [comment]
Hi @colings86!
You said that the fix should be available from version 1.5.3, but it only available in master. I didn't find it in 1.5.3.
